crates/re_viewer/src/store_hub.rs

```diff
@@ -212,37 +221,95 @@ impl StoreHub {
         self.active_application_id = Some(app_id);
     }
 
+    #[inline]
     pub fn active_application_id(&self) -> Option<&ApplicationId> {
         self.active_application_id.as_ref()
     }
 
-    /// Change which blueprint is active for a given [`ApplicationId`]
+    /// Change which blueprint is the default for a given [`ApplicationId`]
     #[inline]
-    pub fn set_blueprint_for_app_id(&mut self, blueprint_id: StoreId, app_id: ApplicationId) {
-        re_log::debug!("Switching blueprint for {app_id} to {blueprint_id}");
-        self.blueprint_by_app_id.insert(app_id, blueprint_id);
+    pub fn set_default_blueprint_for_app_id(
+        &mut self,
+        blueprint_id: &StoreId,
+        app_id: &ApplicationId,
+    ) {
+        re_log::debug!("Switching default blueprint for {app_id} to {blueprint_id}");
+        self.default_blueprint_by_app_id
+            .insert(app_id.clone(), blueprint_id.clone());
+    }
+
+    /// Make blueprint active for a given [`ApplicationId`]
+    ///
+    /// We never activate a blueprint directly. Instead, we clone it and activate the clone.
+    //TODO(jleibs): In the future this can probably be handled with snapshots instead.
+    pub fn make_blueprint_active_for_app_id(
+        &mut self,
+        blueprint_id: &StoreId,
+        app_id: &ApplicationId,
+    ) -> anyhow::Result<()> {
+        let new_id = StoreId::random(StoreKind::Blueprint);
+
+        re_log::debug!(
+            "Cloning {blueprint_id} as {new_id} the active blueprint for {app_id} to {blueprint_id}"
+        );
+
+        // Create the cloned blueprint
+        let blueprint = self
+            .store_bundle
+            .get(blueprint_id)
+            .context("missing blueprint")?;
+        let info = blueprint.store_info().context("missing info")?;
+
+        let mut new_info = info.clone();
+        new_info.store_id = new_id.clone();
+
+        blueprint.store().sort_indices_if_needed();
+
+        let mut new_blueprint =
+            EntityDb::from_info_and_rows(new_info, blueprint.store().to_rows()?)?;
+        // Also copy the data source
+        new_blueprint.data_source = blueprint.data_source.clone();
```

Comment:
This looks messy and somewhat error prone. Maybe better to handle this as a member function?

```suggestion
        let new_blueprint = blueprint.clone_with_new_id(new_id);
```